src/parser/core/modules/PrecastStatus.js

```diff
@@ -14,64 +14,87 @@ export default class PrecastStatus extends Module {
+			if (event.type === 'applybuff' && !statusInfo.hasOwnProperty('stacksApplied')) {
+				// If status applies stacks, check applybuffstack for applying full stacks before considering this the first application of this status
+				this.markStatusAsTracked(statusInfo.id, targetId)
 			}
 
-			if (['removebuff', 'applybuffstack', 'removebuffstack', 'refreshbuff'].includes(event.type)) {
-				const statusId = event.ability.guid
+			if (event.type === 'applybuffstack' && statusInfo.hasOwnProperty('stacksApplied')) {
+				// Determine if this is applying fewer than the max stacks
+				if (event.stack < statusInfo.stacksApplied) {
+					// Synth the precast status event if this applied fewer than max stacks
+					this.fabricateStatusEvent(event, statusInfo)
+				}
+
+				this.markStatusAsTracked(statusInfo.id, targetId)
+			}
 
+			if (['removebuff', 'removebuffstack', 'refreshbuff'].includes(event.type)) {
 				// If it's already been applied, we don't have to worry about it
-				if (this._combatantStatuses[targetId].includes(statusId)) {
+				if (this._combatantStatuses[targetId].includes(statusInfo.id)) {
 					continue
 				}
 
-				// Fab an event and splice it in at the start of the fight
-				events.splice(0, 0, {
-					// Can inherit most of the event data from the current one
-					...event,
-					// Override a few vals
-					timestamp: startTime - 1,
-					type: 'applybuff',
-				})
-
-				// Determine if this buff comes from a known action, fab a cast event
-				const statusInfo = getDataBy(STATUSES, 'id', event.ability.guid)
-				if (statusInfo) {
-					const actionInfo = getDataBy(ACTIONS, 'statusesApplied', statusInfo)
-					// Action found - push it into _combatantActions array if not already there and synthesize a cast event
-					// If action has already been synthesized (and pushed into _combatantActions array), do nothing
-					if (actionInfo && this._combatantActions.indexOf(actionInfo.id) === -1) {
-						this._combatantActions.push(actionInfo.id)
-
-						events.splice(0, 0, {
-							...event,
-							ability: {
-								...event.ability,
-								name: actionInfo.name,
-								abilityIcon: actionInfo.icon.replace('https://xivapi.com/i', '').replace('/', '-'),
-								guid: actionInfo.id,
-							},
-							timestamp: startTime - 2,
-							type: 'cast',
-						})
-					}
-				}
-
-				this._combatantStatuses[targetId].push(statusId)
+				this.fabricateStatusEvent(event, statusInfo)
+				this.markStatusAsTracked(statusInfo.id, targetId)
 			}
 		}
 
-		return events
+		const synthesizedEvents = this._actionsToSynth.concat(this._statusesToSynth)
+		return synthesizedEvents.concat(events)
+	}
+
+	fabricateStatusEvent(event, statusInfo) {
+		// Fab an event and splice it in at the start of the fight
+		this._statusesToSynth.push({
+			// Can inherit most of the event data from the current one
+			...event,
+			// Override a few vals
+			timestamp: this._startTime - 1,
+			type: 'applybuff',
+		})
+
```

Comment:
for the sake of InsaneAccuracy:tm:, is it worth fabricating an `applybuffstack` event after the `applybuff` in the case we're fabbing a stacked event? would replicate fflog's events more closely (i.e. applying an accel mid-combat will generate `[applybuff(accel), applybuffstack(accel, 3)]`

Reply:
Yeah, seems simple enough.  Dun (dun chicken)